lib/openstack/openstack_handle/handle.rb

```diff
@@ -60,9 +63,15 @@ def self.raw_connect(username, password, auth_url, service = "Compute", extra_op
       raise
     end
 
-    def self.auth_url(address, port = 5000, ssl = false)
-      scheme = ssl ? "https" : "http"
-      "#{scheme}://#{address}:#{port}/v2.0/tokens"
+    def self.auth_url(address, port = 5000, scheme = "http")
+      url(address, port, scheme)
+    end
+
+    def self.url(address, port = 5000, scheme = "http", path = "/v2.0/tokens")
+      port = port.to_i
+      uri = URI::Generic.build(:scheme => scheme, :port => port, :path => path)
+      uri.hostname = address
```

Comment:
See #2172 , I'll be changing this to what you'd expect:
`URI::Generic.build(:scheme => scheme, :host => address, :port => port, :path => path)`